lib/server/nahnojiresolver.py

```diff
@@ -15,21 +15,19 @@ def supports(url):
 
 # returns the steam url
 def url(url):
-	#if supports(url):
 	m = _regex(url)
-	if not m == None:
+	if m:
 		ur = m.group('url')
-		if ur.find('.flv') > -1:
-			return [ur]
-		data = util.substr(util.request(ur),'<body style','</a>')
-		pattern = 'href=\"(.+?)\"[^>]+>' 
-		match = re.compile(pattern).findall(data)
-		return [match[0]]
+		if m.group('flv'): return [ur]
+		page = util.parse_html(ur)
+		data = page.select('source[type=video/mp4]')[0]['src']
```

Comment:
Why do you always return the first instance only ? Returning all of them costs nothing.